### ISSUE TITLE
[BUG] [Azure.AI.OpenAI] Validation error at #/dataSources/0/AzureCognitiveSearch/parameters/authentication/APIKey/APIKey: Field required

### ISSUE THREAD

**Issue by thomas-christiansen** [Azure/azure-sdk-for-net#41138]
### Library name and version

Azure.AI.OpenAI beta.12
https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.OpenAI_1.0.0-beta.12/sdk/openai/Azure.AI.OpenAI/README.md

### Describe the bug

All the sudden when I call GetChatCompletionsAsync using the method described in the readme I get this error:
`
Validation error at #/dataSources/0/AzureCognitiveSearch/parameters/authentication/APIKey/APIKey: Field required
`

This just started happening today. I havent made any changes to the code or the search index. When I try to run the reqeust in the Azure AI Studio it works just fine. 

What is going on?

Here is my code
      
            var client = new OpenAIClient(new Uri(_apiBase), new AzureKeyCredential(_apiKey!));

            var primeText = "...";

            AzureCognitiveSearchChatExtensionConfiguration searchExtensionConfig = new()
            {
                SearchEndpoint = new Uri(_searchEndpoint),
                IndexName = _searchIndexName,
                Authentication = new OnYourDataApiKeyAuthenticationOptions(_searchKey),
                DocumentCount = 10,
                Strictness = 1,
                RoleInformation = primeText
            };

            var chatCompletionsOptions = new ChatCompletionsOptions()
            {
                DeploymentName = _deploymentId,
                Messages =  {
                    new ChatRequestSystemMessage(primeText)
                },
                AzureExtensionsOptions = new AzureChatExtensionsOptions()
                {
                    Extensions = { searchExtensionConfig }
                },
                Temperature = 0.8f

            };
         

            var response = await client.GetChatCompletionsAsync(chatCompletionsOptions);

### Expected behavior

I should run without errors as it has been doing until today

### Actual behavior

The error has started to occur - has any thing changed? I'm not seeing a new nuget package, the latest is v12-beta

### Reproduction Steps

The code is in the description

**Comment by thesnake666** [comment]
Unfortunately I have the same error with the same client code and library version.
Also for me this behavior just started happening today!
Please, help us.
Thanks

**Comment by WueF** [comment]
For me exactly the same issue occured today for the first time, without changing anything to code. Library version is exactly the same.

**Comment by freistli** [comment]
I experienced the same issue, it suddenly happened several hours ago.

**Comment by LucGosso** [comment]
Same here, similar code, but version beta11. It started 3 days ago, then it didnt return any result from search first. 2 days ago it worked, yesterday on my important demonstration, it didnt work. Same error, about missingkey.

I guess something changed in azure endpoint?

**Comment by kokuljose** [comment]
Same issue is happening for my two client project for past 5+ hours which is live. Azure.AI.OpenAI version is 1.0.0-beta.12

**Comment by horakmi2** [comment]
Same Issue here

**Comment by sonfer** [comment]
I didn't make any changes to my codes that were working yesterday. But it stopped working properly.

**Comment by LucGosso** [comment]
How do we report this to Azure?

**Comment by LucGosso** [comment]
I pay for services but can report it when its not working, great deal

![image](https://github.com/Azure/azure-sdk-for-net/assets/9716195/aea1e9cf-c956-46e5-bee0-ef7268bf30c1)

**Comment by 126463** [comment]
Same issue since this morning, just before a demo.

**Comment by LucGosso** [comment]
Filed an issue via X and got a free technical ticket, got the answer from Azure support: 

________________________________

Thank you for connecting with us. I have gone through all the details you shared; I would like to thank you for briefly describing the complete issue. To start troubleshooting your issue, I would suggest that can you please roll back to the previous Azure OpenAI Package using dotnet. You can use this command for rolling it back to the beta.6 version.
 
`Command: dotnet add package Azure.AI.OpenAI --version 1.0.0-beta.6`

________________________________

If someone wants to try before me ... let us know

**Comment by kokuljose** [comment]
I have downgraded the system to beta-11, yet the issue remains unresolved. It seems that downgrading to beta 6 will necessitate making corresponding changes to the code, particularly in relation to the message defined in azure.ai.openai. This will require careful consideration and thorough implementation to ensure the successful resolution of the error.

**Comment by brunobmo** [comment]
Same with me. Noticed that stopped working yesterday...

**Comment by claymartin74** [comment]
I have the same issue today... Started happening in production (no deployment).

**Comment by edwardkuo** [comment]
I get the same issue. So nuget version is bug?

**Comment by 126463** [comment]
> I get the same issue. So nuget version is bug?

I think so, everything works fine with Playground - Chat Completions with custom data.

**Comment by MahdiSheikhi** [comment]
I observed that the system wasn't functioning correctly since yesterday. 

However, I've implemented a temporary fix with the following steps:

- Enabled Identity for the Azure OpenAI resource.
- Assigned the Contributor role to the Azure Search Service.
- Modified the authentication type of the client object in the code.

I changed the authentication from using an API key:
 
`Authentication = new OnYourDataApiKeyAuthenticationOptions(myApiKey),
`

to using a System Assigned Managed Identity:

`Authentication = new OnYourDataSystemAssignedManagedIdentityAuthenticationOptions(),
`

These changes have resolved the issue for now.

**Comment by PaulCousinsTTEducation** [comment]
To get the system managed identity workaround to be successful you have to enable "Both" under the keys blade of the search service. I also had to assign both the "Contributor" and "Search Index Data Contributor" roles. Only then did it work for me. This was using the beta.12 version.

**Comment by thesnake666** [comment]
This morning, without any changes in the code from my side, the service is back to work :-)

**Comment by thomas-christiansen** [comment]
I'm getting a different error now: 
'NoneType' object has no attribute 'get'

Whatever that means...

**Comment by brunobmo** [comment]
Still the same APIKey error. No changes today. I will go try the workaround proposed using Identity

**Comment by moremanu** [comment]
Same issue since yesterday. Tried with two different solutions, different approaches same error.

**Comment by freistli** [comment]
The Azure OpenAI service starts getting to normal, but AzureExtensionsContext.Messages doesn't return Citations. I'm going to create another bug

**Comment by LucGosso** [comment]
@freistli yes, i have the same behavior now. No Citations. 

result.Value.Choices[0].Message.AzureExtensionsContext.Messages.Any()//no citation here, broken service

**Comment by freistli** [comment]
> I'm getting a different error now: 'NoneType' object has no attribute 'get'
> 
> This happens when I include the "Assistant" response to the message history. This has not been an issue before, so this looks like another bug that have been introduced. I know this is a preview library, but come on. You could at least keep it in sync with the azure openai service....

I hit the same issue "NoneType" yesterday, but today it backs to normal.

**Comment by freistli** [comment]
> @freistli yes, i have the same behavior now. No Citations.
> 
> result.Value.Choices[0].Message.AzureExtensionsContext.Messages.Any()//no citation here, broken service

@LucGosso I created the new bug here: https://github.com/Azure/azure-sdk-for-net/issues/41241

**Comment by thesnake666** [comment]
Yes, confirm, the result is ok for content but the citations field is null

**Comment by LucGosso** [comment]
Out of nothing all working now as expected, citation too, without code changes. Do we know what caused this service break?

**Comment by thomas-christiansen** [comment]
Can confirm. It is working as it should again. 

I wonder when we can expect a stable version of Azure OpenAI and these components....

**Comment by GraysonQuartech** [comment]
Currently getting this error....
